app/mixins/newuser.js

```diff
@@ -35,17 +36,21 @@ export default Mixin.create(ValidationErrorDisplay, {
   isSaving: false,
   nonStudentMode: true,
 
-  schools: computed('currentUser.model.schools.[]', {
-    get(){
-      return new Promise(resolve => {
-        this.get('currentUser.model').then(user => {
-          resolve(user.get('schools'));
-        });
-      });
-    }
-  }).readOnly(),
+  /**
+   * All schools that the current user has permissions to create users in.
+   * @method schools
+   * @return {Promise<Array>}
+   */
+  async schools() {
```

Comment:
This should still be a computed property, the original was just in an old syntax.